src/uarch.rs

```diff
@@ -0,0 +1,380 @@
+use crate::uarch::Core::{Heterogeneous, Homogenous};
+use crate::Vendor;
+use crate::Vendor::{Intel, Amd};
+
+#[non_exhaustive]
+pub enum CoreArch {
+    // Not including Intel micro-architecture without CPUID suport, for now.
+    // Intel Micro-architectures (with CPUID support)
+    i486,
```

Comment:
there is some naming warning for this

Reply:
variant `i486` should have an upper camel case name

Reply:
Maybe we can make sure we match with information from here https://github.com/intel/perfmon?